opensrp-app/src/main/java/org/smartregister/sync/intent/P2pProcessRecordsService.java

```diff
@@ -0,0 +1,122 @@
+package org.smartregister.sync.intent;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.smartregister.CoreLibrary;
+import org.smartregister.domain.FetchStatus;
+import org.smartregister.domain.db.EventClient;
+import org.smartregister.repository.AllSharedPreferences;
+import org.smartregister.repository.EventClientRepository;
+import org.smartregister.sync.ClientProcessorForJava;
+import org.smartregister.util.Utils;
+
+import java.util.List;
+
+import timber.log.Timber;
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 10/05/2019
+ */
+
+public class P2pProcessRecordsService extends IntentService {
+
+    /**
+     * Creates an IntentService.  Invoked by your subclass's constructor.
+     *
+     * @param name Used to name the worker thread, important only for debugging.
+     */
+    public P2pProcessRecordsService(String name) {
+        super(name);
+    }
+
+    public P2pProcessRecordsService() {
+        super("P2pProcessRecordsService");
+    }
+
+    @Override
+    protected void onHandleIntent(@Nullable Intent intent) {
+        AllSharedPreferences allSharedPreferences = CoreLibrary.getInstance().context().allSharedPreferences();
+
+        if (allSharedPreferences.isPeerToPeerUnprocessedEvents()) {
+            long eventsMaxRowId = allSharedPreferences.getLastPeerToPeerSyncProcessedEvent();
+            EventClientRepository eventClientRepository = CoreLibrary.getInstance().context().getEventClientRepository();
+
+            while (eventsMaxRowId > -1) {
+                EventClientQueryResult eventClientQueryResult = eventClientRepository.fetchEventClientsByRowId(eventsMaxRowId);
+                List<EventClient> eventClientList = eventClientQueryResult.getEventClientList();
+
+                if (eventClientList.size() > 0) {
+                    try {
+                        getClientProcessor().processClient(eventClientList);
```

Comment:
Maybe you can try to check that the ClientProcessor from `org.smartregister.sync.intent.SyncIntentService` is also not running.

Because client Processor should run only in one instance could try to harmonize so that both refer to one instance of ClientProcessor

Reply:
> Because client Processor should run only in one instance could try to harmonize so that both refer to one instance of ClientProcessor

Currently, both will use the same instance of the `ClientProcessorForJava` unless the host application decides not to explicitly set their version of the `ClientProcessorForJava`. Is there something that I am missing here?

> Maybe you can try to check that the ClientProcessor from org.smartregister.sync.intent.SyncIntentService is also not running.

Kindly explain

Reply:
You should have one instance of ClientProcessor running. So make sure the Client Processor ir retrieved in the same way from both services e.g though a getter on the Library. 

That will ensure one object is created and the `synchorized` contract is respected

Reply:
Ok, so this does happen right now. Ama?

Reply:
It can happen if one passes the same instance to both, but presents the bug for passing two instances where one Overrides in one services and forgets to override in the other.

Could you define/retrieve the clientProcessor from the CoreLibrary

Reply:
Do we need to make a breaking change on the `SyncIntentService` that makes the `getClientProcessor` method `private`?

Reply:
The method should be to `CoreLibrary` class as a public getter method and both Intent services to retrieve the `clientProcessor` from the CoreLibrary e.g.

 `  CoreLibrary.getInstance().getClientProcessor().processClient(events);`

Reply:
Cool, just asking this since the client apps are overriding that method so as to provide their own processor